lib/recurly/client.rb

```diff
@@ -68,44 +73,40 @@ def pager(path, **options)
 
     def get(path, **options)
       path = scope_by_site(path, **options)
-      request = HTTP::Request.new(:get, path, nil)
-      faraday_resp = run_request(request, headers)
-      handle_response! request, faraday_resp
-    rescue Faraday::ClientError => ex
-      raise_network_error!(ex)
+      request = Net::HTTP::Get.new path
+      http_response = run_request(request, options)
+      handle_response! request, http_response
     end
 
     def post(path, request_data, request_class, **options)
       request_class.new(request_data).validate!
       path = scope_by_site(path, **options)
-      request = HTTP::Request.new(:post, path, JSON.dump(request_data))
-      faraday_resp = run_request(request, headers)
-      handle_response! request, faraday_resp
-    rescue Faraday::ClientError => ex
-      raise_network_error!(ex)
+      request = Net::HTTP::Post.new path
+      request.set_content_type(JSON_CONTENT_TYPE)
+      request.body = JSON.dump(request_data)
+      http_response = run_request(request, options)
+      handle_response! request, http_response
     end
 
     def put(path, request_data = nil, request_class = nil, **options)
       path = scope_by_site(path, **options)
-      request = HTTP::Request.new(:put, path)
+      request = Net::HTTP::Put.new path
       if request_data
         request_class.new(request_data).validate!
-        logger.info("PUT BODY #{JSON.dump(request_data)}")
-        request.body = JSON.dump(request_data)
+        json_body = JSON.dump(request_data)
+        logger.info("PUT BODY #{json_body}")
+        request.set_content_type(JSON_CONTENT_TYPE)
```

Comment:
It seems the server wants the content-type to be json regardless of whether or not there is a body. Moving this line to line `93` fixes.

Reply:
Makes sense, updated.